marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/extensions/runFlow.sjs

```diff
@@ -19,27 +19,46 @@ const datahub = new DataHub();
 
 //todo flush this out
 function get(context, params) {
-  let flowName = params["flowName"];
+  post(context, params, null);
+}
 
-  let flow = null;
+function post(context, params, input) {
+  let flowName = params.flowName;
 
-  if(!fn.exists(flowName)) {
-      fn.error(null,"RESTAPI-SRVEXERR",  Sequence.from([400, "Bad Request", "Invalid request - must specify a flowName"]));
+  if (!fn.exists(flowName)) {
+    fn.error(null,"RESTAPI-SRVEXERR",  Sequence.from([400, "Bad Request", "Invalid request - must specify a flowName"]));
   }
-  else{
-    flow = datahub.flow.getFlow(flowName);
-    let uris = params['identifier'];
-
+  else {
+    let options = params.options ? JSON.parse(params.options) : {};
+    let jobId = params.jobId || datahub.hubUtils.uuid();
+    let flow = datahub.flow.getFlow(flowName);
+    let query = null;
+    let uris = null;
+    if (params.uri) {
+      uris = datahub.hubUtils.normalizeToArray(params.uri);
+      query = cts.documentUriQuery(uris);
+    } else {
+      query = flow.identifier ? cts.query(flow.identifier) : cts.orQuery([]);
+    }
+    let content = null;
+    if (input) {
+      content = input.toObject();
+    } else {
+      content = {};
+      datahub.hubUtils.queryLatest(function() {
+        for (let doc of cts.search(query, cts.indexOrder(cts.uriReference()))) {
+          content[xdmp.nodeUri(doc)] = doc;
+        }
+      }, flow.sourceDb || datahub.flow.globalContext.sourceDb);
+      uris = Object.keys(content);
```

Comment:
@aebadirad , I thought the collector would run and provide the uris which will be passed to datahub.flow.runFlow() ? Has that changed ?

Reply:
This still supports that, if I'm reading it correctly, but will also handle the case of running it on the server locally if no uri list is provided